src/UCI.cpp

```diff
@@ -145,8 +145,8 @@ namespace {
   // Return the score from the self-play game.
   // Precondition: bh.cur() is not a terminal position.
   int play_one_game(BoardHistory& bh) {
-    auto search = std::make_unique<UCTSearch>(bh.shallow_clone());
     for (int game_ply = 0; game_ply < 450; ++game_ply) {
+      auto search = std::make_unique<UCTSearch>(bh.shallow_clone());
```

Comment:
How about a comment saying we are doing this on purpose to avoid tree reuse? Otherwise it might look like a bug.

Reply:
Done.